lib/transport/abstract-transport.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+var util = require("util");
+var EventEmitter = require('events');
+
+var AbstractTransport = function () {
+	if (this.constructor === AbstractTransport) {
+		throwBecauseAbstract();
```

Comment:
it's a little more work, but I'd prefer the errors thrown to be explicit. For instance, here it would be `cannot instantiate AbstractTransport`, below it would be `Subclass must implement 'write'`. Since we aren't going to export this class, rather just duck type outside implementations, implementors will probably run into these and more direction would be useful